extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -0,0 +1,243 @@
+		$contributorDetails = $this->getContributorsDetails( array_slice( $contributors, 0, 5 ) );
+
+		$this->response->setData( [
+			'topContribsHeaderText' => $this->msg( 'communitypage-top-contributors-week' )->plain(),
+			'yourRankText' => $this->msg( 'communitypage-user-rank' )->plain(),
+			'userContributionsText' => $this->msg( 'communitypage-user-contributions' )
+				->numParams( $userContribCount )
+				->text(),
+			'contributors' => $contributorDetails,
+			'userAvatar' => AvatarService::renderAvatar(
+				$this->getUser()->getName(),
+				AvatarService::AVATAR_SIZE_SMALL_PLUS
+			),
+			'userRank' => 'N',
+			'memberCount' => 'N',
+			'userContribCount' => $userContribCount
+		] );
+	}
+
+	/**
+	 * Set context for topAdmins template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getTopAdminsData() {
+		$topAdmins = CommunityPageSpecialUsersModel::filterGlobalBots(
+			// get all admins who have contributed in the last two years ordered by contributions
+			CommunityPageSpecialUsersModel::getTopContributors( 10, '2 YEAR', true )
+		);
+		$topAdminsDetails = $this->getContributorsDetails( $topAdmins );
+
+		$remainingAdminCount = count( $topAdmins ) - 2;
+
+		$this->response->setData( [
+			'topAdminsHeaderText' => $this->msg( 'communitypage-admins' )->plain(),
+			'otherAdmins' => $this->msg( 'communitypage-other-admins' )->plain(),
+			'admins' => array_slice( $topAdminsDetails, 0, 2 ),
+			'otherAdminCount' => $remainingAdminCount,
+			'haveOtherAdmins' => $remainingAdminCount > 0,
+			'adminCount' => count( $topAdmins ),
+		] );
+	}
+
+	/**
+	 * Set context for recentlyJoined template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getRecentlyJoinedData() {
+		$recentlyJoined = $this->usersModel->getRecentlyJoinedUsers();
+
+		$this->response->setData( [
+			'allMembers' => $this->msg( 'communitypage-view-all-members' )->plain(),
+			'recentlyJoinedHeaderText' => $this->msg( 'communitypage-recently-joined' )->plain(),
+			'members' => $recentlyJoined,
+		] );
+	}
+
+	/**
+	 * Set context for allMembers template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getAllMembersData() {
+		$allMembers = $this->usersModel->getAllMembers();
+
+		$this->response->setData( [
+			'allMembersHeaderText' => $this->msg( 'communitypage-all-members' )->plain(),
+			'admin' => $this->msg( 'communitypage-admin' )->plain(),
+			'joinedText' => $this->msg( 'communitypage-joined' )->plain(),
+			'members' => $allMembers,
+		] );
+	}
+
+	/**
+	 * Set context for recentActivityModule template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	private function getRecentActivityData() {
+		$data = $this->sendRequest( 'LatestActivityController', 'executeIndex' )->getData();
+		$recentActivity = [];
+
+		foreach ($data['changeList'] as $activity) {
```

Comment:
The code still isn't spacey in many places like here. I'd suggest running the PHP formatter from our guidelines repo over the files.